shell/src/main/java/alluxio/cli/docgen/ConfigurationDocGenerator.java

```diff
@@ -206,17 +223,82 @@ public static void writeYMLFile(Collection<? extends PropertyKey> defaultKeys, S
 
   /**
    * Generates the configuration docs.
+   * @param validate validate flag
    */
-  public static void generate() throws IOException {
+  public static void generate(boolean validate) throws IOException {
```

Comment:
note that the generate function can throw an IOException. if the validate flag is true and a temp file is generated, it needs to be deleted even if an exception is thrown, otherwise the git workspace will be dirty. there should be some exception handling to handle this scenario and ensure that the generated file is deleted even if an exception is thrown